src/JsonSchema/Constraints/Factory.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the JsonSchema package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JsonSchema\Constraints;
+
+use JsonSchema\Exception\InvalidArgumentException;
+use JsonSchema\Uri\UriRetriever;
+use JsonSchema\Validator;
+
+/**
+ * Factory for centralize constraint initialization.
+ */
+class Factory
+{
+    /**
+     * @var UriRetriever
+     */
+    protected $uriRetriever;
+
+    /**
+     * @param UriRetriever $uriRetriever
+     */
+    public function __construct(UriRetriever $uriRetriever = null)
+    {
+        if (!$uriRetriever) {
+            $uriRetriever = new UriRetriever();
+        }
+
+        $this->uriRetriever = $uriRetriever;
+    }
+
+    /**
+     * @return UriRetriever
+     */
+    public function getUriRetriever()
+    {
+        return $this->uriRetriever;
+    }
+
+    /**
+     * Create a constraint instance for the given constraint name.
+     *
+     * @param string $constraintName
+     * @return ConstraintInterface|ObjectConstraint
+     * @throws InvalidArgumentException if is not possible create the constraint instance.
+     */
+    public function factory($constraintName)
```

Comment:
I think this might be clearer if called `getConstraintValidator` or something similar. The class is a factory, this is a getter/builder.